src/testIntegration/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_SwitchResponsibleOfficerTest.java

```diff
@@ -0,0 +1,235 @@
+                .body(createResponsibleOfficerSwitchOf(true))
+                .when()
+                .put("/offenders/nomsNumber//responsibleOfficer/switch")
+                .then()
+                .statusCode(400);
+    }
+
+    @Test
+    @DisplayName("Will switch Responsible officer from COM to POM and back again")
+    public void canSwitchResponsibleOfficerToPrisonOffenderManager() {
+        // Given an offender has no prison offender manager
+        assertThat(prisonOffenderManager(getOffenderManagers("G0560UO"))).isEmpty();
+
+        // Then the responsible officer can not be assigned to a POM
+        given()
+                .auth()
+                .oauth2(tokenWithRoleCommunityAndCustodyUpdate())
+                .contentType(APPLICATION_JSON_VALUE)
+                .contentType("application/json")
+                .body(createResponsibleOfficerSwitchOf(false))
+                .when()
+                .put("/offenders/nomsNumber/G0560UO/responsibleOfficer/switch")
+                .then()
+                .statusCode(409);
+
+        // Given an offender has Prisoner Offender Manager
+        allocatePrisonerOffenderManager("G0560UO");
+
+        // AND the COM is currently the Responsible Officer
+        final var offenderManagersBeforeSwitch = getOffenderManagers("G0560UO");
+
+        assertThat(communityOffenderManager(offenderManagersBeforeSwitch).orElseThrow().getIsResponsibleOfficer())
+                .isTrue();
+        assertThat(prisonOffenderManager(offenderManagersBeforeSwitch).orElseThrow().getIsResponsibleOfficer())
+                .isFalse();
+
+        // WHEN I request responsible officer is switched to the POM
+        given()
+                .auth()
+                .oauth2(tokenWithRoleCommunityAndCustodyUpdate())
+                .contentType(APPLICATION_JSON_VALUE)
+                .contentType("application/json")
+                .body(createResponsibleOfficerSwitchOf(false))
+                .when()
+                .put("/offenders/nomsNumber/G0560UO/responsibleOfficer/switch")
+                .then()
+                .statusCode(200);
+
+        // AND when I check who is current Responsible officer currently is
+        final var offenderManagersAfterSwitch = getOffenderManagers("G0560UO");
+
+        // THEN the responsible officer is now set to Prison Offender Manager
+        assertThat(prisonOffenderManager(offenderManagersAfterSwitch).orElseThrow().getIsResponsibleOfficer()).isTrue();
+        assertThat(communityOffenderManager(offenderManagersAfterSwitch).orElseThrow().getIsResponsibleOfficer())
+                .isFalse();
+
+        // AND a CONTACT has been added to show responsible officer has changed
+        final var contacts = getRecentContacts("G0560UO");
+
+        assertThat(Arrays
+                .stream(contacts)
+                .anyMatch(contact -> contact.getContactType().getDescription().equals("Responsible Officer Change"))
```

Comment:
Should we check that we have 0 contacts before, then 1 contact after first RO switch, then 2 contacts after 2nd RO switch?

Reply:
maybe, I opted not to, e.g if the first works not to bother to test again